### ISSUE TITLE
Implement automagic KMS server discovery in pykms_Client

### ISSUE THREAD

**Issue by Matthew-Beckett** [Py-KMS-Organization/py-kms#51]
https://github.com/SystemRage/py-kms/issues/121

**Comment by simonmicro** [comment]
Duplicate of #45

**Comment by Matthew-Beckett** [comment]
This is not a duplicate this is a feature request

**Comment by simonmicro** [comment]
sry

**Comment by edgd1er** [comment]
From my understanding, the py_client is ran to bootstart the local kms server in order to reach 25 activations.
what would be the usage to query random kms servers ? (well, server would be known after a DNS request srv _vlmcs._tcp)

**Comment by simonmicro** [comment]
> py_client is ran to bootstart the local kms server in order to reach 25 activations

Nope, just to trigger the database creation for the sqlite webinterface...

> what would be the usage to query random kms servers ?

Could you elaborate?

**Comment by edgd1er** [comment]
> > py_client is ran to bootstart the local kms server in order to reach 25 activations
> 
> Nope, just to trigger the database creation for the sqlite webinterface...
> 
> > what would be the usage to query random kms servers ?
> 
> Could you elaborate?

Automatic KMS discovery is based on _vlmcs._tcp.domain.tld DNS query. Should you have two or more KMS server on your lan, you can't know for sure what will be the DNS answer, hence what would be the KMS used to activate your client. 

although, I get the point, If you wish to validate kms discovery the py_kms client, that feature is needed.

**Comment by edgd1er** [comment]
#66 is a proposition to implement KMS discovery feature.

**Comment by Matthew-Beckett** [comment]
Closed, feature merged in #66
